proto/xmtpv4/message_api/message_api.proto

```diff
@@ -109,6 +123,26 @@ message QueryEnvelopesResponse {
   repeated GatewayEnvelope envelopes = 1;
 }
 
+message PublishEnvelopeRequest {
+  PayerEnvelope payer_envelope = 1;
+}
+
+message PublishEnvelopeResponse {
+  message PublishEnvelopeSuccess {
+    OriginatorEnvelope originator_envelope = 1;
+  }
+  message PublishEnvelopeFailure {
```

Comment:
Can't we just handle the error cases with GRPC errors and metadata outlining the reason?

Then all the automatically generated metrics are meaningful, and when we have a GRPC gateway endpoint we also get HTTP status codes correctly set.

Reply:
Totally fair, I misremembered the outcome of our previous discussion. Will just remove it